R/simulation.R

```diff
@@ -290,9 +290,27 @@ setMethod(
     # source module metadata and code files, checking version info
     lapply(modules(sim), function(m) {
       md <- moduleMetadata(m, modulePath(sim))
-      if (md$version != packageVersion("SpaDES")) {
+      mVersion <- unlist(md$version)
+      pVersion <- unlist(packageVersion("SpaDES"))
+      maxLength <- max(length(mVersion), length(pVersion))
+      if(length(mVersion) != length(pVersion)){
+        if(length(mVersion) != maxLength){
+          mVersion[(length(mVersion)+1):maxLength] <- 0
+        } else {
+          pVersion[(length(pVersion)+1):maxLength] <- 0
+        }
+      }
+      maxDigits <- max(max(ceiling(log10(mVersion+1)), na.rm = TRUE),
+                       max(ceiling(log10(pVersion+1)), na.rm = TRUE))
+      mVersionNumeric <- mVersion[maxLength]
+      pVersionNumeric <- pVersion[maxLength]
+      for(i in 1:(maxLength-1)){
+        mVersionNumeric <- mVersionNumeric + mVersion[i]*10^((maxLength-i)*maxDigits)
+        pVersionNumeric <- pVersionNumeric + pVersion[i]*10^((maxLength-i)*maxDigits)
+      }
+      if (mVersionNumeric %>>% pVersionNumeric) {
```

Comment:
what is going on here? why not simply change the check in the `if` block to look for `>=` /  `>=`?? Expanding the `if() {}` into an `if () {} else if() {}` or similar would be easier.

This way we can better tailor the messages to the user depending on:

1. module version == spades version : no problem
2. module version < spades version : likely no problem, as spades should be backwards compatible
3. module version > spades version : the user needs to update their version of spades

@eliotmcintire, we had been discussing how to better handle this version checking. care to weigh in?